src/Ssh/Ssh/Common/SshBaseCmdlet.cs

```diff
@@ -550,18 +577,45 @@ protected internal string GetProxyPath()
         protected internal void GetVmIpAddress()
         {
             string _message = "";
-            Ip = this.IpUtils.GetIpAddress(Name, ResourceGroupName, UsePrivateIp, out _message);
+           
+            (string IpAdress, NetworkInterface networkInterface) = this.IpUtils.GetIpAddress(Name, ResourceGroupName, UsePrivateIp,Bastion, out _message);
 
-            if (_message.StartsWith("Unable to find public IP.") && !UsePrivateIp)
```

Comment:
Looks like I'm the culprit who was checking the message to see if the public IP was not found 😆 
Definitely change that, have GetIpAddress return an explicit value to indicate whether it found the IP or not.